packages/api/src/util/sensorCSV.js

```diff
@@ -0,0 +1,278 @@
+/*
+ *  Copyright 2019, 2020, 2021, 2022 LiteFarm.org
+ *  This file is part of LiteFarm.
+ *
+ *  LiteFarm is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+
+// Sensor bulk upload error translation keys
+const sensorErrors = {
+  FILE_ROW_LIMIT_EXCEEDED: 'FARM_MAP.BULK_UPLOAD_SENSORS.VALIDATION.FILE_ROW_LIMIT_EXCEEDED',
+  MISSING_COLUMNS: 'FARM_MAP.BULK_UPLOAD_SENSORS.VALIDATION.MISSING_COLUMNS',
+  EXTERNAL_ID: 'FARM_MAP.BULK_UPLOAD_SENSORS.VALIDATION.EXTERNAL_ID',
+  SENSOR_NAME: 'FARM_MAP.BULK_UPLOAD_SENSORS.VALIDATION.SENSOR_NAME',
+  SENSOR_LATITUDE: 'FARM_MAP.BULK_UPLOAD_SENSORS.VALIDATION.SENSOR_LATITUDE',
+  SENSOR_LONGITUDE: 'FARM_MAP.BULK_UPLOAD_SENSORS.VALIDATION.SENSOR_LONGITUDE',
+  SENSOR_READING_TYPES: 'FARM_MAP.BULK_UPLOAD_SENSORS.VALIDATION.SENSOR_READING_TYPES',
+  SENSOR_DEPTH: 'FARM_MAP.BULK_UPLOAD_SENSORS.VALIDATION.SENSOR_DEPTH',
+  SENSOR_BRAND: 'FARM_MAP.BULK_UPLOAD_SENSORS.VALIDATION.SENSOR_BRAND',
+  SENSOR_MODEL: 'FARM_MAP.BULK_UPLOAD_SENSORS.VALIDATION.SENSOR_MODEL',
+  SENSOR_HARDWARE_VERSION: 'FARM_MAP.BULK_UPLOAD_SENSORS.VALIDATION.SENSOR_HARDWARE_VERSION',
+  SENSOR_ALREADY_OCCUPIED: 'FARM_MAP.BULK_UPLOAD_SENSORS.SENSOR_CLAIM_ERROR.ALREADY_OCCUPIED',
+  SENSOR_DOES_NOT_EXIST: 'FARM_MAP.BULK_UPLOAD_SENSORS.SENSOR_CLAIM_ERROR.DOES_NOT_EXIST',
+  INTERNAL_ERROR: 'FARM_MAP.BULK_UPLOAD_SENSORS.SENSOR_CLAIM_ERROR.INTERNAL_ERROR',
+};
+
+const sensorCsvValidators = [
```

Comment:
I still think that this should probably be passed into the function which parses the csv, instead of being the always called. There may be a scenario where LiteFarm uses a csv upload for something else now that it is already being done in one place, so I think there is a lot of value to making this function more reusable.

Reply:
Maybe if we have a parseCsv function, and then a parseSensorCsv function which acts as a wrapper to the original function and uses this set of validators.

Reply:
Sure